app/src/main/java/app/revanced/manager/ui/component/InstallerDialog.kt

```diff
@@ -0,0 +1,144 @@
+package app.revanced.manager.ui.component
+
+import android.content.pm.PackageInstaller
+import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.Check
+import androidx.compose.material.icons.outlined.ErrorOutline
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import app.revanced.manager.R
+import app.revanced.manager.ui.viewmodel.InstallerViewModel
+import com.github.materiiapps.enumutil.FromValue
+
+private typealias InstallerDialogButtonHandler = ((vm: InstallerViewModel) -> Unit)
+private typealias InstallerDialogButton = @Composable (vm: InstallerViewModel) -> Unit
+
+
+@Composable
+fun InstallerDialog(
+    vm: InstallerViewModel,
+) {
+    val dialogKind = DialogKind.fromValue(vm.installerStatus!!) ?: DialogKind.FAILURE
+
+    AlertDialog(
+        onDismissRequest = {},
```

Comment:
This makes it impossible to dismiss the dialog by clicking outside of it. Is this intentional?

Reply:
It is not intentional, but considering that, I'd keep it this way, because the dialogs have buttons to dismiss them and if you accidentally tap outside of it, it'll be annoying.

Reply:
The dialogs created by this `AlertDialog` overload are usually quite big if I recall correctly, so I don't think people will accidentally click outside of them. All other dialogs in the app are dismissable like this, so not being able to dismiss it like this would be inconsistent. The only exception to this is the first launch auto update settings dialog, because it is absolutely required to read and acknowledge it.